File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int,int> mp;
        for (int val : arr){
            ++mp[val];
        }
        vector<int> ans_arr;
        for (const auto &p : mp){
            ans_arr.push_back(p.second);
        }
        sort(ans_arr.begin(),ans_arr.end());
        for(int i=0;i<ans_arr.size();i++){
            if (i+1<ans_arr.size() && ans_arr[i]==ans_arr[i+1]){
                return false;
            }

        }
        return true;

        
    }
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
from collections import defaultdict
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        hashmap= defaultdict(int)
        for i in arr:
            hashmap[i]+=1
        arr1 = list(hashmap.values())
        arr1.sort()
        for i in range(len(arr1)):
            if (((i+1)<(len(arr1))) and arr1[i]==arr1[i+1]):
                return False
        return True
        
        
        